salt/beacons/inotify.py

```diff
@@ -105,37 +106,42 @@ def __validate__(config):
     ]
 
     # Configuration for inotify beacon should be a dict of dicts
-    log.debug('config {0}'.format(config))
-    if not isinstance(config, dict):
-        return False, 'Configuration for inotify beacon must be a dictionary.'
+    if not isinstance(config, list):
+        return False, 'Configuration for inotify beacon must be a list.'
     else:
-        for config_item in config:
-            if not isinstance(config[config_item], dict):
+        _config = {}
+        list(map(_config.update, config))
+
+        for path in _config.get('files', {}):
```

Comment:
Yes, but your call `.get('files', {})` in the for loop context will create an empty dict if no files are configured which won't iterate because its empty. If we're not iterating but need an iterator, a tuple should suffice. Unless I'm missing something.